mesonbuild/compilers/mixins/islinker.py

```
@@ -94,9 +94,9 @@ def get_allow_undefined_link_args(self) -> T.List[str]:
         raise mesonlib.EnvironmentException(
             'Linker {} does not support allow undefined'.format(self.id))
 
-    def get_pie_link_args(self) -> T.List[str]:
-        m = 'Linker {} does not support position-independent executable'
-        raise mesonlib.EnvironmentException(m.format(self.id))
+    def get_pie_link_args(self, enabled: T.Optional[bool]) -> T.List[str]:
+        """Get linker pie arguments."""
+        return []
```

Same here, removing the exception will make it silently ignore the pie value.